src/include/duckdb/main/external_dependencies.hpp

```diff
@@ -6,17 +6,73 @@
 //
 //===----------------------------------------------------------------------===//
 
+#include "duckdb/common/case_insensitive_map.hpp"
+#include <functional>
+
 #pragma once
 
 namespace duckdb {
 
-enum class ExternalDependenciesType : uint8_t { PYTHON_DEPENDENCY };
+enum class ExternalDependencyItemType : uint8_t { PYTHON_DEPENDENCY };
```

Comment:
Can we remove the `ExternalDependencyItemType` and use dynamic casts instead like the `FunctionData`? This is used exclusively in extensions/clients/etc that the core cannot know about so having an enum in the core can never be exhaustive anyway